packages/core/src/analytics/payment-providers-tracker/payment-providers-tracker.ts

```diff
@@ -0,0 +1,77 @@
+import { AnalyticStepType } from '../step-tracker/analytics-step-tracker';
+
+import PaymentProvidersTrackerService from './payment-providers-tracker-service';
+import { BraintreeConnectTracker } from './trackers';
+import { EventsPayload } from './types';
+
+export default class PaymentProvidersTracker implements PaymentProvidersTrackerService {
+    private braintreeConnectTracker?: BraintreeConnectTracker;
+
+    constructor () {
+        this.braintreeConnectTracker = new BraintreeConnectTracker();
```

Comment:
Can we have a tracker service per provider? Since here we have the `braintreeConnectTracker` which means this could be tied to BT, but in future we could have different `tracker` and then we will end up adding conditional logic possibly.

Reply:
I think we might have trackers for different providers in the future. So I created abstract class `PaymentProvidersTracker` that can call other trackers whenever it needed.

```
this.braintreeConnectTracker = new BraintreeConnectTracker();
this.paypalCommerceConnectTracker = new PaypalCommerceConnectTracker();
this.stripeTracker = new StripeTracker();
```